internal/broadcaster/broadcaster_test.go

```diff
@@ -0,0 +1,129 @@
+package broadcaster_test
+
+import (
+	"container/list"
+	"context"
+	"fmt"
+	"math"
+	"reflect"
+	"testing"
+	"time"
+
+	"log/slog"
+	"os"
+
+	"github.com/bitcoin-sv/arc/internal/broadcaster"
+	"github.com/bitcoin-sv/arc/internal/testdata"
+	"github.com/bitcoin-sv/arc/pkg/keyset"
+	"github.com/libsv/go-bt/v2"
+	"github.com/libsv/go-bt/v2/bscript"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	batchSizeDefault = 20
+	maxInputsDefault = 100
+)
+
+func TestBroadcasterWithPredefinedUTXOs(t *testing.T) {
+	// Set up a logger
+	logger := slog.New(slog.NewTextHandler(os.Stdout, &slog.HandlerOptions{Level: slog.LevelDebug}))
+
+	// Create a KeySet
+	ks1, err := keyset.New()
+	require.NoError(t, err)
+
+	// Define UTXOs
+	utxo1 := &bt.UTXO{
+		TxID:          testdata.TX1Hash[:],
+		Vout:          0,
+		LockingScript: ks1.Script,
+		Satoshis:      1000,
+	}
+	utxo2 := &bt.UTXO{
+		TxID:          testdata.TX2Hash[:],
+		Vout:          0,
+		LockingScript: ks1.Script,
+		Satoshis:      2000,
+	}
+	utxo3 := &bt.UTXO{
+		TxID:          testdata.TX3Hash[:],
+		Vout:          0,
+		LockingScript: ks1.Script,
+		Satoshis:      3000,
+	}
+
+	// Mock UTXO Client
+	utxoClient := &MockUtxoClient{
+		utxos: []*bt.UTXO{utxo1, utxo2, utxo3},
+	}
```

Comment:
Instead of using a self created mock client you can use the generated one: https://github.com/bitcoin-sv/arc/blob/31ff37554515365022b9aaf81aa672e722e29771/internal/broadcaster/mocks/utxo_client_mock.go